patches/tock/07-firmware-protect.patch

```diff
@@ -0,0 +1,350 @@
+diff --git a/boards/components/src/firmware_protection.rs b/boards/components/src/firmware_protection.rs
+new file mode 100644
+index 0000000..5eda591
+--- /dev/null
++++ b/boards/components/src/firmware_protection.rs
+@@ -0,0 +1,67 @@
++//! Component for firmware protection syscall interface.
++//!
++//! This provides one Component, `FirmwareProtectionComponent`, which implements a
++//! userspace syscall interface to enable the code readout protection.
++//!
++//! Usage
++//! -----
++//! ```rust
++//! let crp = components::firmware_protection::FirmwareProtectionComponent::new(
++//!     board_kernel,
++//!     nrf52840::uicr::Uicr::new()
++//! )
++//! .finalize(
++//!     components::firmware_protection_component_helper!(uicr));
++//! ```
++
++use core::mem::MaybeUninit;
++
++use capsules::firmware_protection;
++use kernel::capabilities;
++use kernel::component::Component;
++use kernel::create_capability;
++use kernel::hil;
++use kernel::static_init_half;
++
++// Setup static space for the objects.
++#[macro_export]
++macro_rules! firmware_protection_component_helper {
++    ($C:ty) => {{
++        use capsules::firmware_protection;
++        use core::mem::MaybeUninit;
++        static mut BUF: MaybeUninit<firmware_protection::FirmwareProtection<'static, $C>> = MaybeUninit::uninit();
++        &mut BUF
++    };};
```

Comment:
Even on master, I don't see this macro being used for components.
I tried very quickly to use it but got complains from the compiler because the struct doesn't derive copy.
Will try to have a deeper look later

Reply:
Ok no problem, we can do it in a later PR. This change should be a no-op regarding how the key functions. It's just that this pattern looks a bit risky (although the dangerous part is hidden under `static_init_half!`).

Reply:
I'm fine with a further PR. This will give me time to try to get this reviewed upstream.

Reply:
I added `static_buf!` in https://github.com/tock/tock/pull/1924, and it's already used internally by `static_init!`. However, IIRC using it in components still depends on reaching an agreement on https://github.com/tock/tock/pull/1618.

I'm also fine with deferring the discussion on the pull request that will happen on https://github.com/tock/tock, rather than settling on too many details here that will later be re-discussed on the Tock repo.